src/test/esperanza/finalizationstate_withdraw_tests.cpp

```diff
@@ -92,11 +92,11 @@ BOOST_AUTO_TEST_CASE(process_withdraw_too_early) {
 
   // From epoch end_logout+1 until end_withdraw-1 finalizer can't withdraw.
   // At end_withdraw or later finalizer can withdraw its deposit.
-  uint32_t end_withdraw = end_logout + static_cast<uint32_t>(spy.WithdrawalEpochDelay());
-  BOOST_CHECK_EQUAL(end_withdraw, 21);
+  uint32_t end_withdraw = end_logout + static_cast<uint32_t>(spy.WithdrawalEpochDelay()) + 1;
+  BOOST_CHECK_EQUAL(end_withdraw, 22);
 
   for (uint32_t i = spy.GetCurrentEpoch(); i < end_withdraw; ++i) {
-    if (spy.GetCurrentDynasty() < validator->m_end_dynasty) {
+    if (spy.GetCurrentDynasty() <= validator->m_end_dynasty) {
```

Comment:
I'm confused, at the start of `end_dynasty` we should not vote anymore. This is the dynasty where we actually exit the validator set.

Reply:
at the start of `end_dynasty` finalizer is still voting because of this rule https://github.com/dtr-org/unit-e/blob/master/src/esperanza/finalizationstate.cpp#L1108-L1110

Reply:
That's true, all right then